src/synthesizer/extensions/integrated_spectra.c

```diff
@@ -69,6 +69,92 @@ static double *get_spectra_serial(struct grid *grid_props,
   return spectra;
 }
 
+
+/** Example function , not sure if we're going to go this route
+ * @brief Compute the integrated spectra from the grid weights,
+ * accounting for Doppler shift due to velocity.
+ *
+ * @param grid_props: The grid properties.
+ * @param grid_weights: The grid weights computed from the particles.
+ * @param velocities: The velocities associated with each grid cell.
+ * @param c: The speed of light.
+ *
+ * @return The integrated spectra.
+ */
+static double *get_shifted_spectra_serial(struct grid *grid_props,
```

Comment:
Ah yea, this doesn't work. I knew there was something sticky. The weights are calculated separately for the integrated (if you take a look at the rest of this file you'll see there are some extra functions relative to the particle version). So we would need to associate the velocities to the weights somehow and honestly that's just a nightmare. 

It's conceptually (and probably mechanically) simpler to redirect `generate_lnu` to `generate_particle_lnu` if the shift flag is true.